src/main/java/net/datafaker/service/RandomService.java

```diff
@@ -1,14 +1,24 @@
 package net.datafaker.service;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Random;
+import java.util.Set;
 
 public class RandomService {
+    private static final String WEIGHT_KEY = "weight";
+    private static final String VALUE_KEY = "value";
     private static final char[] HEX_UP = "0123456789ABCDEF".toCharArray();
     private static final char[] HEX_LOWER = "0123456789abcdef".toCharArray();
     private static final Random SHARED_RANDOM = new Random();
     private final Random random;
 
+    private double[] cumulativeWeights;
+    private Object[] values;
+
```

Comment:
Definitively it should be outside of this class
Otherwise it will slow down all other not weight related cases

There should be a weight provider dedicated interface/class instead this very generic

Reply:
I've moved all the logic for the weight selector to the WeightedRandomSelector class. Currently, I’ve placed it in the service package, as it seems to fit best there, though I believe you initially suggested putting it in the providers package?

Additionally, based on the issue description, WeightedRandomSelector appears intended mainly for custom providers. At this point, no provider has data in the format required for this functionality. Should I register it only to be used with custom providers, or do you have a different solution in mind?